linera-core/src/chain_worker/state/guard.rs

```diff
@@ -913,6 +913,18 @@ where
 
             info.requested_pending_message_bundles = messages;
         }
+        if !query.request_sent_certificate_hashes_by_heights.is_empty() {
```

Comment:
I don't think the `if` adds anything here (it's just duplicating the first check of the `for`).

(Empty vectors don't allocate.)

Reply:
Done